src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -113,64 +113,109 @@ It is implemented recursively to ensure type stability.
     return self + row + _interactions(as, rest)
 end
 
+"""
+    trap_potential(add::SingleComponentFockAddress, pot)
+
+Calculate trap potential energy of a single-component address `add`. The (precomputed) 
+potential energy per particle at each point in the lattice is given by the vector `pot`.
+"""
+function trap_potential(add::SingleComponentFockAddress, pot)
+    pe = 0.0
+    for (n,i) in occupied_modes(add)
+        pe += n * pot[i]
+    end
+    return pe
+end
+
+function trap_potential(add::CompositeFS, pot::Matrix)
+    pe = 0.0
+    for (i,c) in enumerate(add.components)
+        pe += trap_potential(c, @view pot[:,i])
+    end
+    return pe
+end
```

Comment:
I was just thinking that this is a very general function that could be re-used in other contexts as well (momentum-space, general orbital basis, etc.).

The code looks like it could be optimised for speed. Eg. currently bounds checking is done inside the for loop. Managing bounds checking outside the loop could produce nice speed-ups making use of `@simd` operations.